### PR TITLE
user profile stuff

### PR DESCRIPTION
**Pull request by GusCaplan**

### CODE REVIEW

File: src/structures/UserProfile.js [hunk 1]
```diff
@@ -56,6 +57,8 @@ class UserProfile {
     for (const connection of data.connected_accounts) {
       this.connections.set(connection.id, new UserConnection(this.user, connection));
     }
+
+    if (data.user && data.user.flags) this.user.flag = Constants.UserFlags[data.user.flags];
```
Comment (Gawdl3y) [inline src/structures/UserProfile.js]: Why is this overwritten here?

File: src/structures/User.js [hunk 2]
```diff
@@ -62,6 +62,12 @@ class User {
      * @type {?Message}
      */
     this.lastMessage = null;
+
+    /**
+     * Flag on the user. One of `STAFF`, `PARTNER`, `HYPESQUAD`
+     * @type {?String}
+     */
+    this.flag = data.flags ? Constants.UserFlags[data.flags] : null;
```
Comment (Gawdl3y) [inline src/structures/User.js]: Aren't the flags a bitfield?

File: src/util/Constants.js [hunk 3]
```diff
@@ -371,6 +371,12 @@ exports.DefaultAvatars = {
   RED: '1cbd08c76f8af6dddce02c5138971129',
 };
 
+exports.UserFlags = {
```
Comment (Gawdl3y) [inline src/util/Constants.js]: Assuming this represents a bitfield, the keys/vals should probably be swapped around

File: src/structures/User.js (hunk 2 above)
Comment (GusCaplan) [inline src/structures/User.js]: nope

File: src/util/Constants.js (hunk 3 above)
Comment (GusCaplan) [inline src/util/Constants.js]: nope

File: src/structures/User.js (hunk 2 above)
Comment (Gawdl3y) [inline src/structures/User.js]: So it's a complete coincidence that the numbers happen to be the first three possible flag values for a bitfield... and it's called `flags`, not `flag`...?

**Pull request closed without merging**